docs/tutorials/mongoid-documents.txt

```diff
@@ -604,56 +604,158 @@ of the following methods:
+are called *dynamic fields*.
+
+To enable dynamic fields, include ``Mongoid::Attributes::Dynamic`` module
+in the model:
 
 .. code-block:: ruby
 
-   class Person
-     include Mongoid::Document
-     include Mongoid::Attributes::Dynamic
-   end
+  class Person
+    include Mongoid::Document
+    include Mongoid::Attributes::Dynamic
+  end
+  
+  bob = Person.new(name: 'Bob', age: 42)
+  bob.name
+  # => "Bob"
 
-When dealing with dynamic attributes, the following rules apply:
+It is possible to use ``field`` declarations and dynamic fields in the same
+model class. Attributes for which there is a ``field`` declaration will be
+treated according to the ``field`` declaration, with remaining attributes
+being treated as dynamic fields.
 
-If the attribute exists in the document, Mongoid will provide you with your standard
-getter and setter methods. For example, consider a person who has an attribute of
-"gender" set on the document:
+Attribute values in the dynamic fields must initially be set by either
+passing the attribute hash to the constructor, mass assignment via
+``attributes=``, mass assignment via ``[]=``, using ``write_attribute``,
+or they must already be present in the database.
 
 .. code-block:: ruby
+  
+  # OK
+  bob = Person.new(name: 'Bob')
+  
+  # OK
+  bob = Person.new
+  bob.attributes = {age: 42}
+  
+  # OK
+  bob = Person.new
+  bob['age'] = 42
+  
+  # Raises NoMethodError: undefined method age=
+  bob = Person.new
+  bob.age = 42
+  
+  # OK
+  bob = Person.new
+  # OK - string access
+  bob.write_attribute('age', 42)
+  # OK - symbol access
+  bob.write_attribute(:name, 'Bob')
+  
+  # OK, initializes attributes from whatever is in the database
+  bob = Person.find('123')
 
-   # Set the person's gender to male.
-   person[:gender] = "Male"
-   person.gender = "Male"
+If an attribute is not present in a particular model instance's attributes
+hash, both the reader and the writer for the corresponding field are not
+defined, and invoking them raises ``NoMethodError``:
 
-   # Get the person's gender.
-   person.gender
+.. code-block:: ruby
+  
+  bob = Person.new
+  bob.attributes = {age: 42}
+  
+  bob.age
+  # => 42
+  
+  # raises NoMethodError
+  bob.name
+  
+  # raises NoMethodError
+  bob.name = 'Bob'
+  
+  # OK
+  bob['name'] = 'Bob'
+  
+  bob.name
+  # => "Bob"
 
-If the attribute does not already exist on the document, Mongoid will not provide you with
-the getters and setters and will enforce normal ``method_missing`` behavior. In this case you
-must use the other provided accessor methods:
-(``[]`` and ``[]=``) or (``read_attribute`` and ``write_attribute``).
+Attributes can always be read using mass attribute access or ``read_attribute``
+(this appies to models not using dynamic fields as well):
 
 .. code-block:: ruby
 
-   # Raise a NoMethodError if value isn't set.
-   person.gender
-   person.gender = "Male"
+  bob = Person.new(age: 42)
+  
+  # OK
```

Comment:
Can you write `- string access` here as well? It makes these examples very easy to read.